### PR TITLE
Authorization doc super users

### PR DESCRIPTION
**Pull request by laidan6000**
### Type of change

- Documentation

### Description

This pull request adds information on designating super users to the Authorization section. It provides documentation for issue #1121 .

I also made a few edits to the existing content, e.g. defining ACLs.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/book/proc-kafka-authorization.adoc [hunk 1]
```diff
@@ -24,6 +24,7 @@ spec:
     # ...
     authorization:
       type: simple
+      superUsers:User:Fred;User:Sam;User:Edward
```
Comment (scholzj) [inline documentation/book/proc-kafka-authorization.adoc]:
```suggestion
      superUsers:
        - CN=fred
        - sam
        - CN=edward
```

File: documentation/book/ref-kafka-authorization.adoc [hunk 2]
```diff
@@ -25,4 +24,20 @@ To enable simple authorization, the `type` field should be set to `simple`.
...
+
+Super users can access all resources in your Kafka cluster regardless of any access restrictions defined in ACLs.
+To designate super users for a Kafka cluster, enter a list of user principles, delimited by semi-colons, in the `superUsers` field.
+
+.An example of designating super users
+[source,yaml,subs="attributes+"]
+----
+# ...
+authorization:
+  superUsers:User:Fred;User:Sam;User:Edward
```
Comment (scholzj) [inline documentation/book/ref-kafka-authorization.adoc]:
```suggestion
  type: simple
  superUsers:
    - CN=fred
    - sam
    - CN=edward
```

File: documentation/book/ref-kafka-authorization.adoc [hunk 3]
```diff
@@ -5,18 +5,17 @@
...
 
-Currently, the only supported authorization method is the Simple authorization.
+You can optionally designate a list of super users in the `superUsers` field.
 
 == Simple authorization
 
-Simple authorization is using the `SimpleAclAuthorizer` plugin.
-`SimpleAclAuthorizer` is the default authorization plugin which is part of Apache Kafka.
-To enable simple authorization, the `type` field should be set to `simple`.
+Simple authorization in {ProductName} uses the `SimpleAclAuthorizer` plugin, the default Access Control Lists (ACLs) authorization plugin provided with Apache Kafka.  ACLs allow you to define which users have access to which resources at a granular level.
```
Comment (scholzj) [inline documentation/book/ref-kafka-authorization.adoc]:
```suggestion
Simple authorization in {ProductName} uses the `SimpleAclAuthorizer` plugin, the default Access Control Lists (ACLs) authorization plugin provided with Apache Kafka. ACLs allow you to define which users have access to which resources at a granular level.
```

File: documentation/book/ref-kafka-authorization.adoc [hunk 4]
```diff
@@ -25,4 +24,20 @@ To enable simple authorization, the `type` field should be set to `simple`.
...
   type: simple
 # ...
-----
\ No newline at end of file
+----
+
+== Super users
+
+Super users can access all resources in your Kafka cluster regardless of any access restrictions defined in ACLs.
+To designate super users for a Kafka cluster, enter a list of user principles, delimited by semi-colons, in the `superUsers` field.
```
Comment (scholzj) [inline documentation/book/ref-kafka-authorization.adoc]: I think we should add the information that the users using TLS authentication are prefixed with the `CN=` prefix.

File: documentation/book/ref-kafka-authorization.adoc [hunk 5]
```diff
@@ -5,18 +5,17 @@
 [id='ref-kafka-authorization-{context}']
 = Authorization
 
-Authorization can be configured using the `authorization` property in the `Kafka.spec.kafka` resource.
-When the `authorization` property is missing, no authorization will be enabled.
-When authorization is enabled it will be applied for all enabled xref:assembly-configuring-kafka-broker-listeners-{context}[listeners].
-The authorization method is defined by the `type` field.
+You can configure authorization for Kafka brokers using the `authorization` property in the `Kafka.spec.kafka` resource.
```
Comment (PaulRMellor) [inline documentation/book/ref-kafka-authorization.adoc]:
```suggestion
You configure authorization for Kafka brokers using the `authorization` property in the `Kafka.spec.kafka` resource.
```

File: documentation/book/ref-kafka-authorization.adoc [hunk 6]
```diff
@@ -25,4 +24,25 @@ To enable simple authorization, the `type` field should be set to `simple`.
...
   type: simple
 # ...
-----
\ No newline at end of file
+----
+
+== Super users
+
+Super users can access all resources in your Kafka cluster regardless of any access restrictions defined in ACLs.
+To designate super users for a Kafka cluster, enter a list of user principles in the `superUsers` field. If a user has TLS Client Authentication enabled, prefix their user principle with `CN=`.
```
Comment (PaulRMellor) [inline documentation/book/ref-kafka-authorization.adoc]: What's a 'user principle'?

Comment (laidan6000) [reply documentation/book/ref-kafka-authorization.adoc]: @ppatierno - Please can you check and confirm? I searched for "user principle" in relation to ACLs but could not find anything relevant. Thank you.

Comment (scholzj) [reply documentation/book/ref-kafka-authorization.adoc]: Technically, the thing prefixed with `CN=` is the common name from the certificate subject. So I would phrase it like this:

```suggestion
To designate super users for a Kafka cluster, enter a list of user principles in the `superUsers` field. 
If a user uses TLS Client Authentication, the username will be the common name from their certificate subject prefixed with `CN=`.
```

File: documentation/book/ref-kafka-authorization.adoc [hunk 7]
```diff
@@ -5,18 +5,17 @@
...
-Currently, the only supported authorization method is the Simple authorization.
+You can optionally designate a list of super users in the `superUsers` field.
 
 == Simple authorization
 
-Simple authorization is using the `SimpleAclAuthorizer` plugin.
-`SimpleAclAuthorizer` is the default authorization plugin which is part of Apache Kafka.
-To enable simple authorization, the `type` field should be set to `simple`.
+Simple authorization in {ProductName} uses the `SimpleAclAuthorizer` plugin, the default Access Control Lists (ACLs) authorization plugin provided with Apache Kafka. ACLs allow you to define which users have access to which resources at a granular level.
+To enable simple authorization, set the `type` field to `simple`.
```
Comment (seanhuck) [inline documentation/book/ref-kafka-authorization.adoc]: @laidan6000 should this instance of simple be capitalized?

Comment (laidan6000) [reply documentation/book/ref-kafka-authorization.adoc]: I'm not sure. What do you think, Paolo? @ppatierno

**Pull request merged**